### PR TITLE
Indexing Jupiter Quotes + AMM Swaps

### PR DESCRIPTION
**Pull request by LukasDeco**
- **feat: initial code for jup quote indexing**
- **feat: enum type and startup code for interval fetch indexing**

### CODE REVIEW

File: packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        if (!relatedIx?.args) return Err({ type: "missing data" });
+        const args = relatedIx.args as {
+          args: {
+            swapType: SwapType;
+            inputAmount: BN;
+            outputAmountMin: BN;
+          };
+        };
+        // index a swap here
+        const swapOrder: OrdersRecord = {
```
Comment (LukasDeco) [inline packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts:104]: @R-K-H is the mapping of fields good here?

File: packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+          marketAcct: marketAcct.pubkey.toBase58(),
+          orderBlock: BigInt(transaction.slot),
+          orderTime: transaction.blockTime,
+          orderTxSig: transaction.txSig,
+          quotePrice: BigInt(args.args.inputAmount.toNumber()),
+          actorAcct: userAcct.pubkey.toBase58(),
```
Comment (R-K-H) [inline packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts:110]: Do we use base58 on here to store in the db (I've just noticed this) I THINK they're varchar 44 which is pubkey and string...

File: packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        const args = relatedIx.args as {
+          args: {
+            swapType: SwapType;
+            inputAmount: BN;
+            outputAmountMin: BN;
+          };
+        };
+        // index a swap here
+        const swapOrder: OrdersRecord = {
+          marketAcct: marketAcct.pubkey.toBase58(),
```
Comment (R-K-H) [inline packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts:105]: String I think

File: packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts (hunk 2 above)
Comment (LukasDeco) [reply packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts:110]: I saw Decartes using base58 a lot and I decided to look into and basically it returns the same string as ToString() does but the implementation is more likely to stay where as ToString could change perhaps in the future. That's why I started using it. In any case, it is actually the same address.

File: packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts (hunk 3 above)
Comment (LukasDeco) [reply packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts:105]: check my above comment on this. This method may be "better"

Comment (LukasDeco) [reply packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts:105]: interesting one: https://solana.stackexchange.com/questions/3514/publickey-tostring-vs-publickey-tobase58

File: packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts (hunk 2 above)
Comment (R-K-H) [reply packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts:105]: Perfect

File: packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts (hunk 3 above)
Comment (R-K-H) [reply packages/indexer/src/indexers/amm-market/amm-market-instruction-indexer.ts:100]: lulz

**Review by R-K-H — APPROVED** [review]
Loads here, looks good, the DB changes and update to get it in makes sense. Like we discussed we've certainly got optimizations to work out if needs be.

**Pull request merged**